src/Renci.SshNet/Common/Pipe.cs

```diff
@@ -0,0 +1,47 @@
+﻿namespace Renci.SshNet.Common
+{
+    using System;
+    using System.IO;
+
+    /// <summary>
+    /// A generic pipe to pass through data.
+    /// </summary>
+    internal class Pipe : IDisposable
+    {
```

Comment:
In Java, the PipeInputStream takes a PipeOutputStream as argument.
There's no **Pipe** class.
For example:
http://tutorials.jenkov.com/java-io/pipes.html